### PR TITLE
NAS-106999 / 12.1 / Human-readable error for deleting used cloud credential

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

File: src/middlewared/middlewared/plugins/cloud_sync.py [hunk 1]
```diff
@@ -611,6 +611,9 @@ async def do_delete(self, id):
         """
         Delete Cloud Sync Credentials of `id`.
         """
+        if await self.middleware.call("cloudsync.query", [["credential", "=", id]]):
+            raise CallError("This credential is used")
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/cloud_sync.py:615]: Should we use errno.EBUSY?

@Mrt134 any thoughts in the error message?

Comment (themylogin) [reply src/middlewared/middlewared/plugins/cloud_sync.py:615]: @Mrt134 ping

Comment (yocalebo) [reply src/middlewared/middlewared/plugins/cloud_sync.py:615]: Is there any way to specify which cloud task the credential is being used by?

Comment (yocalebo) [reply src/middlewared/middlewared/plugins/cloud_sync.py:615]: That way you could say something like `This credential is in use by: blah`. If not, then the verbiage you have is fine.

**Pull request merged**